File: charArrayAndString/palindrom.cpp
#include<iostream>
#include<cstring>
#include<algorithm>
using namespace std;
void palindrome(string text){
	string tempT = text;
	reverse(tempT.begin(), tempT.end());
	if(tempT == text){
		cout << "Is palindrom";
	}
	else{
		cout << "Not palindrom";
	}
}
int main(){
	cout << "Enter a text = ";
	string text;
	getline(cin, text);
	palindrome(text);
	return 0;
}

File: charArrayAndString/palindrom.java
package springIntro;

import java.util.*;

public class palindrom {

	static String reverse(String text) {
		StringBuilder str = new StringBuilder(text);
		str.reverse();
		return str.toString();
	}
	
	static void palindrome(String text) {
		String tempT = reverse(text);
		if(tempT.equals(text) == true) {
			System.out.println("Is palindrome");
		}
		else {
			System.out.println("Not palindrome");
		}
	}
	
	public static void main(String[] args) {
		String text;
		Scanner in = new Scanner(System.in);
		System.out.printf("Enter a text = ");
		text = in.next();
		palindrome(text);
	}

}
